src/bin/merchant/main.rs

```diff
@@ -277,6 +277,24 @@ async fn dispatch_channel(
         .await?;
     }
 
+    // The channel has not reacted to a customer posting a mutual close transaction on chain
+    // The condition is
+    // - the contract is closed, but
+    // - the channel status is `PendingMutualClose`
+    if contract_state.status()? == ContractStatus::Closed
+        && channel.status == ChannelStatus::PendingMutualClose
```

Comment:
This seems to leave some options open that shouldn't happen, but nevertheless...
- I believe this code is called when there is an actual change on the contract? Is it possible that a contract changes to ContractStatus::Closed, but ChannelStatus is not PendingMutualClose? Due to an error or a race condition. Did we miss the opportunity to finalize the mutual close or is there a retry?
- Is the other situation possible, PendingMutualClose, but not yet closed on the contract? The idea is that it will still appear, but is there any chance that it doesn't get resolved?

Reply:
Fixed